akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -63,23 +64,7 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
         val builder = PropsData.newBuilder
           .setClazz(props.clazz.getName)
           .setDeploy(deployProto(props.deploy))
-        props.args map serialize foreach builder.addArgs
-        props.args.map { a ⇒
-          val argClassName =
-            if (a == null) "null"
-            else {
-              val className = a.getClass.getName
-              if (scala212OrLater && a.getClass.isInstanceOf[Serializable] && a.getClass.isSynthetic &&
-                className.contains("$Lambda$")) {
-                // The serialization of the parameters is based on passing class name instead of
-                // serializerId and manifest as we usually do. With Scala 2.12 the functions are generated as
-                // lambdas and we can't use that load class from that name when deserializing.
-                classOf[Serializable].getName
-              } else
-                className
-            }
-          builder.addClasses(argClassName)
```

Comment:
Isn't this and also the `args` still needed, so that an old node can deserialize.

Using the `SerializedMessage propsArgs` is of course a nice design, but for backwards compatibility I think you have to reuse the existing lists `args` and `classes`. Adding another list with `serializerIds`, which will be used by new nodes (if it's populated).